code/extensions/DashboardSearchExtension.php

```diff
@@ -79,11 +80,24 @@ function doDashboardSearch() {
 						$searchResults->push(ArrayData::create(array(
 							'Results' => $searchPanel->forTemplate()
 						)));
+						if ($results->count() === 1 && count($searchResults) === 1) {
+							$singleSearchResultItem = $results->first();
+						} else {
+							$singleSearchResultItem = NULL;
+						}
 					}
 				}
 			}
 		}
 
+		if ($singleSearchResultItem) {
+			if ($singleSearchResultItem->config()->dashboard_automatic_search_redirect) {
+				if ($searchResultCMSLink = $singleSearchResultItem->SearchResultCMSLink) {
```

Comment:
I'm not sure if we should use that syntax to call our CMS Link function. I know in the template etc we would have it like that but because ours is specifically a function and is in PHP code, I think that we should have it like `$singleSearchResultItem->getSearchResultCMSLink()` instead of `$singleSearchResultItem->SearchResultCMSLink`.